### PR TITLE
refactor: replace char_u with char

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

File: src/nvim/buffer_defs.h [hunk 1]
```diff
@@ -326,7 +326,7 @@ typedef struct argentry {
  */
 typedef struct {
   char_u *tb_buf;          // buffer for typed characters
-  char_u *tb_noremap;      // mapping flags for characters in tb_buf[]
+  char *tb_noremap;             // mapping flags for characters in tb_buf[]
```
Comment (zeertzjq) [inline src/nvim/buffer_defs.h:329]: I don't this it make sense to change this to `char`. Better use `uint8_t`.

**Comment by justinmk** [comment]
merge conflicts

File: src/nvim/getchar.c [hunk 2]
```diff
@@ -2047,12 +2047,11 @@ static int handle_mapping(int *keylenp, bool *timedout, int *mapdepth)
         // - Partly match: mlen == typebuf.tb_len
         keylen = mp->m_keylen;
         if (mlen == keylen || (mlen == typebuf.tb_len && typebuf.tb_len < keylen)) {
-          char_u *s;
           int n;
 
           // If only script-local mappings are allowed, check if the
           // mapping starts with K_SNR.
-          s = typebuf.tb_noremap + typebuf.tb_off;
+          char_u *s = (char_u *)typebuf.tb_noremap + typebuf.tb_off;
```
Comment (zeertzjq) [inline src/nvim/getchar.c:2054]:
```suggestion
          uint8_t *s = typebuf.tb_noremap + typebuf.tb_off;
```

File: src/nvim/option.c [hunk 3]
```diff
@@ -4613,30 +4613,30 @@ void buf_copy_options(buf_T *buf, int flags)
...
       buf->b_bkc_flags = 0;
-      buf->b_p_gp = empty_option;
-      buf->b_p_mp = empty_option;
-      buf->b_p_efm = empty_option;
-      buf->b_p_ep = empty_option;
-      buf->b_p_kp = empty_option;
-      buf->b_p_path = empty_option;
-      buf->b_p_tags = empty_option;
-      buf->b_p_tc = empty_option;
+      buf->b_p_gp = (char *)empty_option;
```
Comment (bfredl) [inline src/nvim/option.c:4618]: seems we are changing most usages of `empty_option` here, maybe we should change that one to `char *` already.

File: src/nvim/arglist.c [hunk 4]
```diff
@@ -244,7 +244,7 @@ static void get_arglist(garray_T *gap, char *str, int escaped)
 /// "fnames[fcountp]".  When "wig" is true, removes files matching 'wildignore'.
 ///
 /// @return  FAIL or OK.
-int get_arglist_exp(char_u *str, int *fcountp, char ***fnamesp, bool wig)
+int get_arglist_exp(char *str, int *fcountp, char ***fnamesp, bool wig)
```
Comment (zeertzjq) [inline src/nvim/arglist.c:247]: Redundant cast on line 252

**Pull request merged**